This patch sets the OS variable according to the CHOST given by the environment,
thus enabling cross-compilation.

--- configure.orig	2007-04-12 23:51:53.000000000 +0200
+++ configure	2007-04-13 00:03:10.000000000 +0200
@@ -45,6 +45,21 @@
 CFG_CURSES_HEADER=ncurses
 CFG_CURSES_LIB=ncurses
 
+case "$CHOST" in
+  *-darwin*)
+	OS=Darwin
+	;;
+  *-linux*)
+	OS=Linux
+	;;
+  *-freebsd*)
+	OS=FreeBSD
+	;;
+  *-solaris*)
+	OS=SunOS
+	;;
+esac
+
 # Operating system defaults
 [ "$OS" != "" ] || OS=`uname`
 case $OS in
